src/main/java/com/aventstack/extentreports/Report.java

```diff
@@ -190,10 +190,7 @@ private synchronized void collectRunInfo() {
                 test.getExceptionInfoList().forEach(x -> exceptionContextBuilder.setExceptionContext(x, test));
             
             if (test.hasChildren())
-                test.getNodeContext().getAll().forEach(x -> {
-                    copyNodeAttributeInfoToAttributeContext(x);
-                    copyNodeExceptionInfoToExceptionContext(x);
-                });
+                test.getNodeContext().getAll().forEach(this::copyNodeAttributeInfoToAttributeContext);
```

Comment:
The 2 methods:

```
copyNodeAttributeInfoToAttributeContext
copyNodeExceptionInfoToExceptionContext
```

were not sharing the code because one points to the test's attributes and the other for exception.  Attributes of the test are author, category etc.  Exception isn't a test's attribute because it is not something that you can assign, but it occurs due to an event, so its more of a runtime binding. 

We can change the method name so its more clear to the user that here the test is computing all attributes and runtime events..